From 50f9e671feeb43bb2c18ee0a986c69027a4cbd34 Mon Sep 17 00:00:00 2001
From: Jean Trivelly <jean.trivelly@intel.com>
Date: Mon, 25 Feb 2013 18:03:41 +0100
Subject: [PORT FROM MAIN] wifi: Add Freq info in STATUS command in CLI.

BZ: 89448

Add Freq info in STATUS command in CLI.

Change-Id: I78c822f27b11a8b487c170f21a9a8e14a040ec60
Orig-Change-Id: Ic5440df77c727bdbedabbf2d7d0ddfc191257cbe
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: yes
Signed-off-by: Jean Trivelly <jean.trivelly@intel.com>
---
 wpa_supplicant/ctrl_iface.c |   23 +++++++++++++++++++++++
 1 files changed, 23 insertions(+), 0 deletions(-)

diff --git a/wpa_supplicant/ctrl_iface.c b/wpa_supplicant/ctrl_iface.c
index 913f4b7..349a6b1 100644
--- a/wpa_supplicant/ctrl_iface.c
+++ b/wpa_supplicant/ctrl_iface.c
@@ -1439,6 +1439,7 @@ static int wpa_supplicant_ctrl_iface_status(struct wpa_supplicant *wpa_s,
 	end = buf + buflen;
 	if (wpa_s->wpa_state >= WPA_ASSOCIATED) {
 		struct wpa_ssid *ssid = wpa_s->current_ssid;
+		struct wpa_bss *bss = wpa_s->current_bss;
 		ret = os_snprintf(pos, end - pos, "bssid=" MACSTR "\n",
 				  MAC2STR(wpa_s->bssid));
 		if (ret < 0 || ret >= end - pos)
@@ -1512,6 +1513,28 @@ static int wpa_supplicant_ctrl_iface_status(struct wpa_supplicant *wpa_s,
 			if (ret < 0 || ret >= end - pos)
 				return pos - buf;
 			pos += ret;
+
+			/* Add Freq */
+			switch (ssid->mode) {
+			case WPAS_MODE_INFRA:
+				if (bss) {
+					ret = os_snprintf(pos, end - pos, "freq=%d MHz\n",
+							  bss->freq);
+				}
+				break;
+			case WPAS_MODE_IBSS:
+			case WPAS_MODE_AP:
+			case WPAS_MODE_P2P_GO:
+				ret = os_snprintf(pos, end - pos, "freq=%d MHz\n",
+						  ssid->frequency);
+				break;
+			default:
+				ret = 0;
+				break;
+			}
+			if (ret < 0 || ret >= end - pos)
+				return pos - buf;
+			pos += ret;
 		}
 
 #ifdef CONFIG_AP
-- 
1.7.4.1

